### PR TITLE
[REEF-443] Create a Papers section on the website

### PR DESCRIPTION
**Pull request by zerg-junior**
JIRA:

  [REEF-443](https://issues.apache.org/jira/browse/REEF-443)

Pull Request:

  Closes #

### CODE REVIEW

**Comment by zerg-junior** [comment]
This pull request is for discussion. **Do not merge now**.  

Please check the paper list for correctness and completeness. The readable version is available as a [rich diff](https://github.com/apache/reef/pull/1086/files?short_path=dad2bfd#diff-dad2bfd29d34fd452619e2698a1ec5e8).

.bib files are [slides](https://issues.apache.org/jira/browse/REEF-1509) are currently missing because I don't where to link them to. 

@markusweimer  Can you please have a look ?

**Comment by zerg-junior** [comment]
The AppVeyor build failure comes from [TestKMeansOnLocalRuntimeWithGroupCommunications()](https://ci.appveyor.com/project/ApacheSoftwareFoundation/reef/build/805-master/tests), which is unrelated to this PR

**Comment by markusweimer** [comment]
Can you please copy the bib files from my website? Also, we should find more stable references for the PDFs, e.g. from the ACM library.

**Comment by zerg-junior** [comment]
> Can you please copy the bib files from my website

Will do.

> we should find more stable references for the PDFs, e.g. from the ACM library.

AFAIK, ACM requires paid access.

**Comment by zerg-junior** [comment]
Ok, I've made a directory with bib files within the `resources` directory. Does anyone know how I should reference it from markdown, i.e., from `publications.md` ?  The relative paths seem not to be resolved correctly.

**Comment by markusweimer** [comment]
Turns out we should be [fine with copying ACM articles onto our website](http://authors.acm.org/main.html):

> Authors can post the accepted, peer-reviewed version prepared by the author-known as the "pre-print"-to the following sites, with a DOI pointer to the Definitive Version of Record in the ACM Digital Library.
> [...]
> On any non-commercial repository or aggregation that does not duplicate ACM tables of contents, i.e., whose patterns of links do not substantially duplicate an ACM-copyrighted volume or issue. Non-commercial repositories are here understood as repositories owned by non-profit organizations that do not charge a fee for accessing deposited articles and that do not sell advertising or otherwise profit from serving articles.

I believe we are covered by that.

**Comment by zerg-junior** [comment]
I uploaded the pdfs. 
@markusweimer , @tcNickolas Can you please have a look ?

**Comment by tcNickolas** [comment]
The links to files work when I generate site locally.

File: pom.xml [hunk 1]
```diff
@@ -311,6 +311,9 @@ under the License.
                             <!-- The below are binary data files used in tests -->
                             <exclude>Org.Apache.REEF.Examples/ConfigFiles/*</exclude>
                             <exclude>Org.Apache.REEF.Tang.Tests/simpleConstructorJavaProto.bin</exclude>
+							
+							<!-- Bibliography info for papers about REEF -->
```
Comment (tcNickolas) [inline pom.xml]: Please use spaces instead of tabs for indentation.

File: website/src/site/markdown/publications.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Publications
```
Comment (tcNickolas) [inline website/src/site/markdown/publications.md]: Is proper word for this "Publications" or "Papers"? JIRA used "Papers"

File: website/src/site/markdown/publications.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+The main paper on REEF is:
```
Comment (tcNickolas) [inline website/src/site/markdown/publications.md]: JIRA says "We should also provide the bibtex reference for people to use when citing REEF." - should we mention citing in this paragraph?

File: website/src/site/markdown/publications.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+**REEF: Retainable Evaluator Execution Framework**, Markus Weimer et al. *ACM SIGMOD Industrial Track, 2015*. [`[pdf]`](papers/2015-SIGMOD.pdf)[`[bib]`](bib/reef.bib)
+
+
+Other publications, with most recent at the top:
```
Comment (tcNickolas) [inline website/src/site/markdown/publications.md]: Do we need to say "with most recent at the top" if each paper has a year associated with it? The phrase looks a bit weird to me

**Comment by tcNickolas** [comment]
I've done a pass

File: website/src/site/markdown/publications.md (hunk 2 above)
Comment (zerg-junior) [inline website/src/site/markdown/publications.md]: I feel they are largely synonyms in informal speech.

**Comment by tcNickolas** [comment]
LGTM, I'll merge

**Pull request closed without merging**